edb/server/protocol/auth_ext/oauth.py

```diff
@@ -0,0 +1,155 @@
+#
+# This source file is part of the EdgeDB open source project.
+#
+# Copyright 2016-present MagicStack Inc. and the EdgeDB authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+import http
+import datetime
+import base64
+import immutables
+
+from typing import Type, TypeVar, overload
+from jwcrypto import jwt, jwk
+
+from edb import errors
+from edb.server.config import ops
+
+from . import github
+
+
+# TODO get this from edb.server.config.ops directly?
```

Comment:
nah :stuck_out_tongue:

```suggestion
```

Reply:
Technically there is no barrier.

`SettingsMap` is defined under `if TYPE_CHECKING` in `edb/server/config/ops.py` and not exposed in `edb/server/config/__init__.py`, so I assume it's not intended for being imported but rather a local shortcut. Strictly speaking it should be `Mapping[str, SettingValue]` for explicitness I think, so copying the shortcut complies with this explicitness.